dexbot/strategies/relative_orders.py

```diff
@@ -87,7 +87,7 @@ def __init__(self, *args, **kwargs):
 
         # Set external price source, defaults to False if not found
         self.external_feed = self.worker.get('external_feed', False)
-        self.external_price_source = self.worker.get('external_price_source', None)
+        self.external_price_source = self.worker.get('external_price_source', 'gecko')
```

Comment:
I don't like this change: it means if user have `external_feed: true` in the config, but no `external_price_source:`, this will make an implicit assumption that the user wants "gecko". Instead, the strategy should throw an error to the user asking them to explicitly set the price source or correct the settings otherwise.

Reply:
There isn't a way around this unless you also undo the GUI work that has been implemented. The original code, as I wrote it specified the "none" option but Joelva took it out the option in order to satisfy his GUI needs.  So for now, this is the best possible compromise. I'm not changing unless we refactor the whole thing and I did not sign up to fix that

Reply:
Can you describe what exact problem this change supposed to solve? As I see it's nothing to do with original #445 issue.

Reply:
prevented throwing the index error, which is the subject of this open issue.

Reply:
If you really don't want this merge, then we can just drop it instead of wasting time here on a one line issue.

Reply:
How did you tested? I tested and it **doesn't** prevent index error because it happens because of lack of validation in whiptail. That's why I'm asking. I don't have a goal to insult you or argue.

Reply:
@bitfag can you provide information how did you test it and still got the index error? Are you referring to the fix you did in #501? If the problem was that user can set something that is not in the list, the default of `gecko` can stay as it is, since the strategy checks if the worker is using external feed or not.

Reply:
@joelvai checkout any version prior to #501 fix, apply suggested change, then try to reproduce as described here https://github.com/Codaone/DEXBot/issues/445#issuecomment-467801441